aten/src/ATen/core/jit_type.h

```diff
@@ -1372,11 +1372,7 @@ struct CAFFE2_API ClassType : public Type {
   // Create a class type with name `name` and its methods stored in `cu`.
   static ClassTypePtr create(
       QualifiedName qualifiedName,
-      std::shared_ptr<CompilationUnit> cu);
-
-  // Create a type representing a Module,
-  // These do not have methods, and are not globally registered
-  static ClassTypePtr createModuleType(std::shared_ptr<CompilationUnit> module);
+      std::shared_ptr<CompilationUnit> cu, bool is_module = false);
```

Comment:
this `is_module` flag is a suspicious to me, since it implies that we are branching somewhere on whether something is a module (and so modules are not quite first class actually)

Reply:
Ah, seems like it is used for ONNX…

Reply:
Discussed in person: ModuleType will be introduced as subtype of ClassType in a future comment. This means that the ClassType has additional module-like behavior (e.g. it has a list of "parameter" objects)